src/transformStateWithClones.js

```diff
@@ -1,13 +1,33 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = Object.assign({}, state);
+  const arr = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const type = actions[i].type;
+
+    if (type === 'clear') {
+      for (const el in newState) {
+        delete newState[el];
+      }
```

Comment:
The `clear` action should create an entirely new empty state object instead of deleting properties one by one from the existing state object. This would be more efficient and aligns with the task's requirement of not mutating the original state or its clones.